conjecture-rust/src/engine.rs

```diff
@@ -504,13 +501,13 @@ where
         Ok(())
     }
 
-    fn execute(&mut self, buf: &DataStream) -> Result<(bool, TestResult), LoopExitReason> {
+    fn execute(&mut self, buf: &DataStreamSlice) -> Result<(bool, TestResult), LoopExitReason> {
```

Comment:
Do we have a practice of creating custom `Result` types in this project?

Reply:
We already have `StepResult` - I think it makes sense to generalize it a bit and cover cases like this one